cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,495 @@
+			categoryinfo.Category = CategoryID1
+		} else if compareCategory(&Category1NoUID0, &percontainerSCC, CategoryID1NoUID0) {
+			tnf.ClaimFilePrintf("Testing if pod belongs to category1NoUID0 ")
+			categoryinfo.Category = CategoryID1NoUID0
+		} else if compareCategory(&Category2, &percontainerSCC, CategoryID2) {
+			categoryinfo.Category = CategoryID2
+		} else if compareCategory(&Category3, &percontainerSCC, CategoryID3) {
+			categoryinfo.Category = CategoryID3
+		} else {
+			categoryinfo.Category = CategoryID4
+		}
+		// after building the containerSCC need to check to which category it is
+		ContainerList = append(ContainerList, categoryinfo)
+	}
+	return ContainerList
+}
+
+func checkContainCategory(addCapability []corev1.Capability, categoryAddCapabilities []string) bool {
+	for _, ncc := range addCapability {
+		return stringhelper.StringInSlice(categoryAddCapabilities, string(ncc), true)
+	}
+	return len(addCapability) > 0
+}
+
+func CheckPod(pod *provider.Pod) []PodListcategory {
+	var containerSCC ContainerSCC
+	containerSCC.HostIPC = NOK
+	if pod.Spec.HostIPC {
+		containerSCC.HostIPC = OK
+	}
+	containerSCC.HostNetwork = NOK
+	if pod.Spec.HostNetwork {
+		containerSCC.HostNetwork = OK
+	}
+	containerSCC.HostPID = NOK
+	if pod.Spec.HostPID {
+		containerSCC.HostPID = OK
+	}
+	containerSCC.SeLinuxContextPresent = NOK
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContextPresent = OK
+	}
+	containerSCC.AllVolumeAllowed, containerSCC.HostDirVolumePluginPresent = AllVolumeAllowed(pod.Spec.Volumes)
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUserPresent = OK
+	} else {
+		containerSCC.RunAsUserPresent = NOK
+	}
+	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.FSGroup != nil {
+		containerSCC.FsGroupPresent = OK
+	} else {
+		containerSCC.FsGroupPresent = NOK
+	}
+	return checkContainerCategory(pod.Spec.Containers, containerSCC, pod.Name, pod.Namespace)
+}
+
+//nolint:gocyclo,funlen
+func compareCategory(refCategory, containerSCC *ContainerSCC, id CategoryID) bool {
+	result := true
+	tnf.ClaimFilePrintf("Testing if pod belongs to category %s", &id)
+	// AllVolumeAllowed reports whether the volumes in the container are compliant to the SCC (same volume list for all SCCs).
+	// True means that all volumes declared in the pod are allowed in the SCC.
+	// False means that at least one volume is disallowed
+	if refCategory.AllVolumeAllowed == containerSCC.AllVolumeAllowed {
+		tnf.ClaimFilePrintf("AllVolumeAllowed = %s - OK", containerSCC.AllVolumeAllowed)
+	} else {
+		result = false
+		tnf.ClaimFilePrintf("AllVolumeAllowed = %s but expected >=<=%s -  NOK", containerSCC.AllVolumeAllowed, refCategory.AllVolumeAllowed)
+	}
+	// RunAsUserPresent reports whether the RunAsUser Field is set to something other than nil as requested by All SCC categories.
```

Comment:
Add additional comments:
// The runAsUser range can be specified in the SCC itself. If not, it is specified in the namespace, see
// https://docs.openshift.com/container-platform/4.11/authentication/managing-security-context-constraints.html#security-context-constraints-pre-allocated-values_configuring-internal-oauth
// runAsUser:
//  type: MustRunAsRange
//  uidRangeMin: 1000
//  uidRangeMax: 2000